bool user_control
int8 action
